### PR TITLE
Make the shit less shitty

### PR DESCRIPTION
**Pull request by marrek13**

### CODE REVIEW

**Review by eupsumup — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+## Description
 
+Our merchant has a drone flying in the field and spraying fertilizer to protect their crops.
+Their field is divided in sectors of 10 x 10. Our merchant can control this Drone via REST by commands:
 
-# How build the project
+* Turn left or right (response example I am on [2, 3] watching on North (or West))
+* Go forward (response Now I am at [2, 3] Or can not do it [9, 9])
+And they can ask for the drone’s status:
+* Where are you? (response [2, 3])
```
Comment (olga-casian) [inline README.md:11]: are examples in task supposed to be bad practice too? to me this is too confusing. at least the requirements should look fine

File: README.md [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
 
+## Task
+We ask you to identify and fix as many issues as possible in the code we provided. 
+You should also pay attention to make notes with all the recommendations for the person who developed this solution. 
 
+### Constraints
+* The drone can look only in 4 directions (North, South, West, East)
```
Comment (olga-casian) [inline README.md:18]: imho we should remove 'only' as it is limiting too much mental model of implementation and suggest not to focus on future changes

File: README.md [hunk 3]
```diff
@@ -1,18 +1,37 @@
...
 
 The project could be run by the following command:
 
-- java -jar drone_challenge-0.0.1-SNAPSHOT.jar -cp DroneChallengeApplication
-
-To run the above command you need to be in the same folder of the jar file.
+`./gradlew bootRun`
 
-# Documentation
+# Recommendations
```
Comment (olga-casian) [inline README.md:35]: we should be more specific with our expectations. if this is write _todo for general development_, then it should be it. if it's _steps to make it prod ready_ - let's put it

File: src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java [hunk 4]
```diff
@@ -13,45 +13,37 @@
...
-            case "F":
-                drone.getDroneState().getDirection().moveForward();
-                break;
-            default:
-                break;
+        switch (direction) {
+            case "R" -> drone.getDroneState().getDirection().turnRight();
+            case "L" -> drone.getDroneState().getDirection().turnLeft();
+            case "F" -> drone.getDroneState().getDirection().moveForward();
+            default -> { System.out.println("Error happened"); }
```
Comment (olga-casian) [inline src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java:46]: I would still have a huge temptation to rewrite the whole terrible nested structure 😬

File: README.md (hunk 1 above)
Comment (yonatankarp) [reply README.md:11]: +1 for that, I would suggest at least to show what we're expecting as a response (e.g. a JSON example)

File: README.md [hunk 5]
```diff
@@ -1,18 +1,37 @@
...
+* The drone can look only in 4 directions (North, South, West, East)
+* The drone can only turn to neighbouring directions (e.g., if the drone is facing North it
+can turn only to West or East)
+
+## Delivery
+Please copy the template repository. We ask for you to return this task to us no more
+than 7 days after the day that you receive it. We recommend to not spend more than 4 hours in total.
+Once you are happy with your output, please share a private GitHub repository of the code
+with the following account: **mg-code-reviewer**
+Good luck and thank you for your time and consideration in being our next SumUpper!
```
Comment (yonatankarp) [inline README.md:27]: I find it a bit more clear - but let me think if you have other opinions 🙏 

```suggestion
## Delivery
Please create your own private repository from this template repository. We ask for
 you to return this task to us no more than 7 days after the day that you receive it.
 We recommend not spending more than 4 hours in total.
Once you are happy with your output, please share the GitHub repository of the code
with the following account: `mg-code-reviewer`
Good luck and thank you for your time and consideration in being our next SumUpper!
```

File: README.md [hunk 6]
```diff
@@ -1,18 +1,37 @@
...
 
 # How run the project
 
 The project could be run by the following command:
 
-- java -jar drone_challenge-0.0.1-SNAPSHOT.jar -cp DroneChallengeApplication
-
-To run the above command you need to be in the same folder of the jar file.
+`./gradlew bootRun`
```
Comment (yonatankarp) [inline README.md:33]: Nitpick, I would suggest putting this in a structured way such as:

````shell
./gradlew bootRun
````

File: src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java (hunk 4 above)
Comment (yonatankarp) [reply src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java:46]: I belive that this is kind of the point 😅

File: src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java [hunk 7]
```diff
@@ -13,45 +13,37 @@
...
-            case "L":
-                drone.getDroneState().getDirection().turnLeft();
-                break;
-            case "F":
-                drone.getDroneState().getDirection().moveForward();
-                break;
-            default:
-                break;
+        switch (direction) {
+            case "R" -> drone.getDroneState().getDirection().turnRight();
```
Comment (yonatankarp) [inline src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java:43]: Did we intentionally add a bug here that the command will fall through all cases? if so I find it a bit too confusing TBH

Comment (marrek13) [reply src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java:43]: Evgeny was doing it so I don't know, but if there's a bug we should fix it. Could you just commit to this branch with all your fixes to save time?

Comment (yonatankarp) [reply src/main/java/com/sumup/drone_challenge/api/controller/DroneController.java:43]: Oh interesting, I didn't know that Java supports switch-expression - seems like the code is fine as it is 😅

File: README.md (hunk 3 above)
Comment (marrek13) [reply README.md:37]: Done

File: README.md (hunk 1 above)
Comment (marrek13) [reply README.md]: I made a fix, please check it out and fix if you see any issues :pray:

**Review by yonatankarp — APPROVED** [review]

**Pull request merged**